### DISCUSSION TITLE
Different path flags for the database and cache

### DISCUSSION THREAD

**Discussion by ipaqsa** [aquasecurity/trivy#4467] [Ideas]
### Description

It would be nice to make two flags instead of the cache-dir flag, one for the database and the other for the cache(fanal). So that we can place the database in the system directory /var/lib/trivy/ and the cache is loaded to the user in /home/user/.cache

### Target

None

### Scanner

None

**Comment by khanh59** [comment]
Hi maintainers,

I got below logs trying to run 2 trivy processes to scan 2 big images in parallel. It seems the ladder process has a bottleneck and could progress only if the former has released the cache dir (finished or canceled).
<img width="939" alt="log" src="https://github.com/aquasecurity/trivy/assets/33947701/6c79ada5-09a3-43d4-99bc-5bb4f1f2f660">
Things worked well when I explicitly define another cache dir for each process. But this leads to the need to redownload database, which occupying more spaces.
I assume that separating the flags to explicitly define optional --db-dir flag (db to be read) and --cache-dir (cache to be written) can enhance trivy to be run in parallel effectively.

I am looking forward for your thought on this. Thank you in advance.

**Comment by ipaqsa** [comment]
Unfortunately this won't help, trivy's db is bbolt, and it doesn't support multithreading. 

> Hi maintainers,
> 
> 
> 
> I got below logs trying to run 2 trivy processes to scan 2 big images in parallel. It seems the ladder process has a bottleneck and could progress only if the former has released the cache dir (finished or canceled).
> 
> <img width="939" alt="log" src="https://github.com/aquasecurity/trivy/assets/33947701/6c79ada5-09a3-43d4-99bc-5bb4f1f2f660">
> 
> Things worked well when I explicitly define another cache dir for each process. But this leads to the need to redownload database, which occupying more spaces.
> 
> I assume that separating the flags to explicitly define optional --db-dir flag (db to be read) and --cache-dir (cache to be written) can enhance trivy to be run in parallel effectively.
> 
> 
> 
> I am looking forward for your thought on this. Thank you in advance.
